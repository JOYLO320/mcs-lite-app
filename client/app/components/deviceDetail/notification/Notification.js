import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import Transition from 'react-motion-ui-pack';
import { IconClose } from 'mcs-lite-icon';
import MobileDetect from 'mobile-detect';
import { P, Button, Notification as MLNotification } from 'mcs-lite-ui';
import { Fixed, RightWrapper, IconWrapper } from './styled-components';
import { switchToMobilePathname } from './utils';

const md = new MobileDetect(window.navigator.userAgent);

class Notification extends React.Component {
  static propTypes = {
    // React-intl i18n
    getMessages: PropTypes.func.isRequired,
  }
  state = { show: md.mobile() }
  onClose = () => this.setState({ show: false });
  render() {
    const { getMessages: t } = this.props;
    const { onClose } = this;

    return this.state.show && (
      <Fixed>
        <Transition
          component={false}
          appear={{ opacity: 0.5, translateY: -40 }}
          enter={{ opacity: 1, translateY: 0 }}
        >
          <MLNotification key="notification">
            <P>{t('switchQuestion')}</P>

            <RightWrapper>
              <Link to={switchToMobilePathname}>
                <Button>{t('switchToMobile')}</Button>
              </Link>
              <IconWrapper onClick={onClose}><IconClose /></IconWrapper>
            </RightWrapper>
          </MLNotification>
        </Transition>
      </Fixed>
    );
  }
}

export default Notification;
