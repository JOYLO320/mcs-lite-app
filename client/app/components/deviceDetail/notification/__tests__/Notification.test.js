import React from 'react';
import { shallow } from 'enzyme';
import R from 'ramda';
import Notification from '../Notification';

it('should redner <Notification /> when it is on server-side', () => {
  const wrapper = shallow(
    <Notification getMessages={R.identity} />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('should redner <Notification /> when it is mobile mode', () => {
  const wrapper = shallow(
    <Notification getMessages={R.identity} />,
  );

  wrapper.setState({ show: true });

  expect(wrapper).toMatchSnapshot();
});

it('should handle onClose', () => {
  const wrapper = shallow(
    <Notification getMessages={R.identity} />,
  );

  // Before onClose
  wrapper.setState({ show: true });
  expect(wrapper.state('show')).toBe(true);

  // After onClose
  wrapper.instance().onClose();
  expect(wrapper.state('show')).toBe(false);
});
