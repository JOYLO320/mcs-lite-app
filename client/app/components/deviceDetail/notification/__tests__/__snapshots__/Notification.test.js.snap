// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should redner <Notification /> when it is mobile mode 1`] = `
<styled.div>
  <Transition
    appear={
      Object {
        "opacity": 0.5,
        "translateY": -40,
      }
    }
    component={false}
    enter={
      Object {
        "opacity": 1,
        "translateY": 0,
      }
    }
    leave={
      Object {
        "opacity": 0,
      }
    }
    onEnter={[Function]}
    onLeave={[Function]}
    runOnMount={true}
  >
    <Notification>
      <P>
        switchQuestion
      </P>
      <styled.div>
        <Button
          active={false}
          block={false}
          component="button"
          kind="primary"
          onClick={[Function]}
          round={false}
          size="normal"
          square={false}
        >
          switchToMobile
        </Button>
        <styled.div
          onClick={[Function]}
        >
          <IconClose
            size="1em"
          />
        </styled.div>
      </styled.div>
    </Notification>
  </Transition>
</styled.div>
`;

exports[`should redner <Notification /> when it is on server-side 1`] = `null`;
